src/bin/chronos-splits-editor/main.rs

```diff
@@ -0,0 +1,646 @@
+    }
+}
+
+#[rustfmt::skip]
+fn edit_start_timer_at_view(s: &mut Cursive) {
+    let globals = s.user_data::<GlobalState>().unwrap();
+
+    let formatter = NoneWrapper::new(SegmentTime::new(), "");
+    let name = formatter.format(globals.splits_editor.offset()).to_string();
+
+    let menu = PaddedView::lrtb(0, 0, 1, 0,
+        ListView::new()
+            .child(
+                "Name",
+                PaddedView::lrtb(0, 0, 0, 1,
+                    EditView::new()
+                        .content(name)
+                        .on_submit(save_start_timer_at),
+                ),
+            )
+            .resized(AtLeast(40), AtLeast(10)),
+    );
+
+    let dialog =
+        Dialog::around(menu)
+            .title("edit game name")
+            .button("close", |s| {
+                s.pop_layer();
+            });
+
+    s.add_layer(dialog);
+}
+
+fn save_start_timer_at(s: &mut Cursive, value: &str) {
+    let globals = s.user_data::<GlobalState>().unwrap();
+
+    match globals.splits_editor.parse_and_set_offset(value) {
+        Ok(_timespan) => (),
+        Err(error) => return show_error(s, &error.to_string()),
+    }
+
+    let formatter = SegmentTime::new();
+    let value = formatter.format(globals.splits_editor.offset()).to_string();
+
+    let mut button = s
+        .find_name::<Button>("details_start_timer_at_edit_view")
+        .unwrap();
+
+    button.set_label(value);
+    match s.pop_layer() {
+        _ => (),
+    }
+}
+
+#[rustfmt::skip]
+fn edit_attempts_view(s: &mut Cursive) {
+    let globals = s.user_data::<GlobalState>().unwrap();
+
+    let name = globals.splits_editor.attempt_count();
+
+    let menu = PaddedView::lrtb(0, 0, 1, 0,
+        ListView::new()
+            .child(
+                "Name",
+                PaddedView::lrtb(0, 0, 0, 1,
+                    EditView::new()
+                        .content(name.to_string())
+                        .on_submit(save_attempts),
+                ),
+            )
+            .resized(AtLeast(40), AtLeast(10)),
+    );
+
+    let dialog =
+        Dialog::around(menu)
+            .title("edit game name")
+            .button("close", |s| {
+                s.pop_layer();
+            });
+
+    s.add_layer(dialog);
+}
+
+fn save_attempts(s: &mut Cursive, value: &str) {
+    let globals = s.user_data::<GlobalState>().unwrap();
+
+    match globals.splits_editor.parse_and_set_attempt_count(value) {
+        Ok(_timespan) => (),
+        Err(error) => return show_error(s, &error.to_string()),
+    }
+
+    let mut button =
+        s.find_name::<Button>("details_attempts_edit_view").unwrap();
+
+    button.set_label(value);
+    match s.pop_layer() {
+        _ => (),
+    }
+}
+
+#[rustfmt::skip]
+fn main_splits(s: &mut Cursive) {
+    let details = Panel::new(update_details_view(s))
+        .title("details")
+        .with_name("details_panel");
+
+    let splits_title = PaddedView::lrtb(0, 0, 0, 1,
+        LinearLayout::horizontal()
```

Comment:
minor style nit: i would have split this into multiple assignments for readability, but as this is how the examples encourage you to do it, its up to you what you do here. feel free to ignore this one

Reply:
I'll refactor it - it will help remove the skip formats